src/Suave.Tests/Types.fs

```diff
@@ -57,6 +57,11 @@ let httpBinding (_ : SuaveConfig) =
       let binding = HttpBinding.create HTTP IPAddress.Loopback 80us
       let actual = binding.uri "" "" |> sprintf "%O"
       Expect.equal actual "http://127.0.0.1/" "uri"
+
```

Comment:
The System.Net.Uri object handles that. After parsing the URI string it will provide both escaped and unescaped data. I can write a couple of tests to check for double encoding/decoding issues and make sure the expectations are explicit.